app/jobs/search_keyword_job.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class SearchKeywordJob < ApplicationJob
+  queue_as :scrapers
+
+  def perform(search_result_id)
+    search_result = SearchResult.find(search_result_id)
+    html_code = Search::SearchService.new(
+      keyword: search_result.keyword,
+      search_engine: search_result.search_engine
+    ).search
+
+    raise IcRailsHueyToby::Errors::ClientServiceError unless html_code
+
+    search_result.update html_code: html_code, status: :completed
+  rescue Faraday::ConnectionFailed, IcRailsHueyToby::Errors::ClientServiceError, NotImplementedError
```

Comment:
If these are the expected errors, we could also refactor it using the `rescue_from` block: https://edgeguides.rubyonrails.org/active_job_basics.html#exceptions (I forgot about this)

```ruby
rescue_from IcRailsHueyToby::Errors::ClientServiceError do |_exception|
    mark_as_failed(search_result)
end
```

How about replacing `NotImplementedError` with `IcRailsHueyToby::Errors::ClientServiceError` as it could simplify the code? In addition, `Faraday::ConnectionFailed` is also replaced in #34

Reply:
I've simplified the exceptions to only use `IcRailsHueyToby::Errors::ClientServiceError` 🙏 

Is it really necessary to use `rescue_from`? If I understand correctly, it's defined outside of `def perform`

I'm a bit confused where it's supposed to get the `search_result` from:

```
  rescue_from IcRailsHueyToby::Errors::SearchServiceError do |_exception, search_result|
    mark_as_failed(search_result)
  end
```

Reply:
It's not compulsory. It was just an idea as it is part of the DSL of ActiveJob.